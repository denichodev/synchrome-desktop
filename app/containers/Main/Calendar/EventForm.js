import React, { Component } from 'react';
import { Field, reduxForm } from 'redux-form';
import moment from 'moment';
import DatePicker from 'react-datepicker';

import styles from './EventForm.css';

class EventForm extends Component {
  renderTextField = field => {
    // const { meta: { touched, error } } = field;
    // const className = `form-group ${touched && error ? 'has-danger' : ''}`;
    const { label, name, input, placeholder, disabled } = field;

    return (
      <div className="form-group">
        <label htmlFor={name}>
          {label}
        </label>
        <input
          type="text"
          className="form-control"
          placeholder={placeholder}
          disabled={disabled}
          {...input}
        />
      </div>
    );
  };

  renderDatepicker = field => {
    const { input, label, name } = field;
    console.log('input datepicker', input);

    return (
      <div className="form-group">
        <label htmlFor={name}>
          {label}
        </label>
        <DatePicker
          selected={input.value}
          onChange={input.onChange}
          className="form-control"
        />
      </div>
    )
  }

  getNumberList = max => {
    let array = [];
    for (let i = 1; i <= max; i++) {
      array.push(i);
    };
    return array;
  }

  render() {

    return (
      <form>
        <Field
          label="Event Name"
          name="name"
          placeholder="Enter your event name here"
          component={this.renderTextField}
        />
        <div className="row">
          <div className="col-md-6">
            <Field
              label="Start Date"
              name="start"
              component={this.renderDatepicker}
            />
          </div>
          <div className="col-md-6">
            <Field
              label="End Date"
              name="end"
              component={this.renderDatepicker}
            />
          </div>  
        </div>
      </form>
    )
  }
}

const formOptions = {
  form: 'EventsNewForm',
  destroyOnUnmount: false
};

export default reduxForm(formOptions)(EventForm);
