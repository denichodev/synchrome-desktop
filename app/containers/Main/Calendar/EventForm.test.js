import React from 'react';
import configureStore from 'redux-mock-store';
import EventForm from './EventForm';

const mockStore = configureStore();

describe('Event form ', () => {
  it('match its snapshot', () => {
    const component = shallow(<EventForm />);
    expect(component).toMatchSnapshot();
  });

  it('test', () => {
    const component = shallow(<EventForm store={mockStore()} />);
    const instance = component.dive().instance();

    console.log(instance.renderForm);
  })
})
