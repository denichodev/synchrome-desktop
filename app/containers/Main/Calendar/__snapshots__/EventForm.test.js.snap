// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event form  match its snapshot 1`] = `
<Connect(Form(EventForm))
  destroyOnUnmount={false}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="EventsNewForm"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
/>
`;
